### ISSUE TITLE
Some questions when training on my dataset

### ISSUE THREAD

**Issue by soul-M-42** [pit-ray/SPADE-pix2pix-for-Anime#2]
I like this repository and tried to train on a full-body portrait dataset. Here are two problems with the output of network:
1. The generated images looks similar tones. This problem also appeared when I train on other types of GANs, so maybe its' not the problem with the code but my dataset(
2. The disappearance of details. The generated images have the correct outline, but the face and clothing details disappeared. I checked the generator code and wonder if there's a skip-connection like in U-Net. As far as I know, this structure is used to improve the effect of details in the pix2pix paper.
![tile_img-573](https://user-images.githubusercontent.com/37969642/122022873-9bd65d80-cdf9-11eb-875b-658e54f8c44e.png)

Just a beginner in ML, どうもありがとう  :)

**Comment by pit-ray** [comment]
I have interested in another result. Thanks.

1. The problem may be called Mode Collapse. Generally, there are strong restrictions in the Conditional GAN like pix2pix. Therefore, their latest spaces tend to be poor expressions. The problem has been tackled by many researchers. For example, Qi Mao. et al. preprint in [arXiv:1903.05628, 2019(v6)](https://arxiv.org/abs/1903.05628) added the diversity term into loss function to solve it.
  <br>However, this method could not solve in my dataset. The code exists in [diversity_loss.py](https://github.com/pit-ray/SPADE-pix2pix-for-Anime/blob/master/diversity_loss.py). Please try it in your dataset. By the way, I alleviated this problem by tuning the hyperparameters of the model against my 500 images, but the method is very hard.

2. As you say, the first proposed GAN-based [pix2pix](https://arxiv.org/abs/1611.07004) architecture adapts U-Net with skip connection. However, in [pix2pixHD](https://arxiv.org/abs/1711.11585) or [GauGAN](https://arxiv.org/abs/1903.07291) which were proposed later, context information is passed to the back layer by Residual Block well known in ResNet. Additionally, GauGAN also injects the semantic label in each layer by normalization called SPADE. Therefore, I think your problem is not the appearance of skip connections, but rather the stagnation of learning due to Mode Collapse.

Thanks.

**Comment by soul-M-42** [comment]
Thanks a lot :)
